### PR TITLE
Modify actions in search page triggering a page reload

### PR DESCRIPTION
**Pull request by edmundoa**
## Description
Many actions in the search page trigger a full page reload, either because they use links to navigate to another page, or because they submit a form to the server.

That was needed in previous versions of the web interface (prior 2.0) but now can be replaced by a combination of react-router and a refresh of the reflux stores involved in the search. This has the advantage of avoiding the page reload, avoiding asking the server for the web interface assets (as described in #2488), and also avoiding flashes with the login page (as described in #2770).

This PR takes care of modifying some of those actions to avoid page reloads when:
- Executing searches
- Using search pagination
- Changing histogram resolution
- Selecting saved searches
- Changing message sorting

This PR intentionally does not take care of:
- Refactoring the code around searches (specially `SearchStore`), as it is too risky for a bugfix release. This is something we can move into 2.2
- Avoiding page reload when using surrounding searches. The main problem here is to reset the search view, e.g. scroll to top, reset page number, and collapse expanded message. Doing those things would require more changes that I don't think are worth pursuing right now. We can take care of fixing it in a follow-up task

## Motivation and Context
The changes were motivated by #2488, and they also fix that issue.

## How Has This Been Tested?
I have checked that these search elements still work after the changes:
- [X] Widgets use right search params
- [X] Save searches use right search params
- [X] CSV export uses right search params (but didn't solve the issue described in #2795)
- [X] Field analyzers use right params
- [X] Searching from sources tab works
- [X] Searching in streams works
- [X] Search live-loading works
- [X] Show surrounding messages works
- [X] Sorting works
- [X] Field selection works

### CODE REVIEW

File: graylog2-web-interface/src/components/search/SearchResult.jsx [hunk 1]
```diff
@@ -144,6 +144,12 @@ const SearchResult = React.createClass({
                          permissions={this.props.permissions} searchInStream={this.props.searchInStream} />
       );
     }
+
+    let loadingIndicator;
+    if (this.props.loadingSearch) {
+      loadingIndicator = <div className="loading-indicator"><Spinner text="Updating search results..."/></div>;
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/search/SearchResult.jsx]: I think it makes sense to extract this into a separate component straight away, that we can reuse in different places too, including moving its styling from the global stylesheet to a local one.

File: graylog2-web-interface/src/stores/search/SearchStore.ts [hunk 2]
```diff
@@ -332,14 +333,18 @@ class SearchStore {
         var searchURLParams = this.getOriginalSearchURLParams();
         searchURLParams = searchURLParams.set("width", this.width);
         searchURLParams = searchURLParams.set(param, value);
-        URLUtils.openLink(this.searchBaseLocation("index") + "?" + Qs.stringify(searchURLParams.toJS()), false);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/stores/search/SearchStore.ts]: As `URLUtils.openLink` is not used anymore, can you please remove it?

**Comment by dennisoelkers** [comment]
Besides those two remarks, it looks good to me.

File: graylog2-web-interface/src/components/search/SearchResult.jsx (hunk 1 above)
Comment (edmundoa) [inline graylog2-web-interface/src/components/search/SearchResult.jsx]: Makes sense, will take care of it ✅

File: graylog2-web-interface/src/stores/search/SearchStore.ts (hunk 2 above)
Comment (edmundoa) [inline graylog2-web-interface/src/stores/search/SearchStore.ts]: ✅

**Pull request merged**